Update docs about sdk client configuration options (#4826)
camunda/camunda-docs: docs/apis-tools/spring-zeebe-sdk/configuration.md

## Additional configuration options

For a full set of configuration options, see [CamundaClientConfigurationProperties.java](https://github.com/camunda/camunda/blob/main/clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/zeebe/spring/client/properties/CamundaClientProperties.java).

### Auth

> Here you find alternatives to authenticate with the cluster

Reviewer: ```suggestion
Authenticate with the cluster using the following alternative methods:
```
Rewording suggestion

#### Username & Password

> You can also authenticate with the cluster through a simple authentication with username and password

Reviewer: ```suggestion
You can authenticate with the cluster using username and password authentication.
```

```yaml
camunda:
  client:
    mode: self-managed
    auth:
      username: <your username>
      password: <your password>
```

#### Keystore & Truststore

> You can also authenticate with the cluster through java's key- and truststore facilities

Reviewer: ```suggestion
You can authenticate with the cluster using Java's Keystore and Truststore.
```
Rewording suggestion

```yaml
camunda:
  client:
    mode: self-managed
    auth:
      keystore-path: <your keystore path>
      keystore-password: <your keystore password>
      keystore-key-password: <your keystore key password>
      truststore-path: <your truststore path>
      truststore-password: <your truststore password>
```

### Zeebe

> Here you find further zeebe specific configuration options

Author: ```suggestion
You can use the following Zeebe-specific additional configuration options:
```

> Here you find further zeebe specific configuration options

Reviewer: You can use the following Zeebe-specific additional configuration options:

      message-time-to-live: PT2H
```

#### Max message size

> A custom maxMessageSize allows the client to receive larger or smaller responses from Zeebe. Technically, it specifies the maxInboundMessageSize of the gRPC channel (default 4MB):

Reviewer: ```suggestion
A custom `maxMessageSize` allows the client to receive larger or smaller responses from Zeebe. Technically, it specifies the `maxInboundMessageSize` of the gRPC channel (default 4MB):
```
Code tags

      max-message-size: 4194304
```

#### Max metadata size

> A custom maxMetadataSize allows the client to receive larger or smaller response headers from Camunda:

Reviewer: ```suggestion
A custom `maxMetadataSize` allows the client to receive larger or smaller response headers from Camunda:
```
Code tag

      rest-address: http://localhost:8080
```

#### Defaults and Overrides

> You can define defaults and overrides for all supported configuration options for a workers.

Reviewer: ```suggestion
You can define defaults and overrides for all supported configuration options for a worker.
```
Fix typo

      rest-address: http://localhost:8080
```

#### Deployment

> @TODO not sure if this is used ?

Reviewer 2: It seems not used, so we can remove it
